app/controllers/datasets_controller.rb

```diff
@@ -20,10 +20,11 @@ def show
   end
 
   def preview
-    @preview = RestClient.get(preview_url(params[:file_id]))
-    render json: @preview
-  rescue
-    render json: { "error": "No preview available" }, status: 404
+    @preview = JSON.parse(RestClient.get(preview_url(params[:file_id])))
+    @content_type = @preview.fetch('content', {}).fetch('type', '')
+    @content_type = @content_type.upcase
+  rescue => e
+    logger.warn("Error while displaying preview: #{e}")
   end
```

Comment:
Does that rescue result in a 500 page?

Reply:
No - it results in a 'preview can't be displayed' page.